From ca31b3c54cd9d138be7d335989a8cb081ab28e12 Mon Sep 17 00:00:00 2001
From: Aleksei Bavshin <alebastr89@gmail.com>
Date: Sun, 4 Jun 2023 05:50:36 -0700
Subject: [PATCH] Store persistent data in the greetd user home.

There are several reasons to diverge from the upstream preference:
 - the location is consistent with our greetd configuration
 - no need to copy the tmpfiles.d workaround for cache directory
   ownership on OSTree-based systems
 - SELinux blocks write access to the `/var/cache/tuigreet` and I'm not
   interested in adding new rules to the greetd-selinux policy
---
 src/info.rs | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/info.rs b/src/info.rs
index 12b75d0..b3e9e66 100644
--- a/src/info.rs
+++ b/src/info.rs
@@ -21,10 +21,10 @@ use crate::{
   Greeter,
 };
 
-const LAST_USER_USERNAME: &str = "/var/cache/tuigreet/lastuser";
-const LAST_USER_NAME: &str = "/var/cache/tuigreet/lastuser-name";
-const LAST_COMMAND: &str = "/var/cache/tuigreet/lastsession";
-const LAST_SESSION: &str = "/var/cache/tuigreet/lastsession-path";
+const LAST_USER_USERNAME: &str = "/var/lib/greetd/tuigreet-lastuser";
+const LAST_USER_NAME: &str = "/var/lib/greetd/tuigreet-lastuser-name";
+const LAST_COMMAND: &str = "/var/lib/greetd/tuigreet-lastsession";
+const LAST_SESSION: &str = "/var/lib/greetd/tuigreet-lastsession-path";
 
 const DEFAULT_MIN_UID: u16 = 1000;
 const DEFAULT_MAX_UID: u16 = 60000;
-- 
2.45.1

